src/commands/usages/index.ts

```diff
@@ -0,0 +1,181 @@
+import fs from 'fs'
+import { uniqBy } from 'lodash'
+import minimatch from 'minimatch'
+import { Flags } from '@oclif/core'
+import { lsFiles } from '../../utils/git/ls-files'
+import { parseFiles } from '../../utils/usages/parse'
+import Base from '../base'
+import { File, JSONMatch, LineItem, VariableReference } from './types'
+import ClientNameFlag, { getClientNames } from '../../flags/client-name'
+import MatchPatternFlag, { getMatchPatterns } from '../../flags/match-pattern'
+import VarAliasFlag, { getVariableAliases } from '../../flags/var-alias'
+import ShowRegexFlag, { showRegex } from '../../flags/show-regex'
+import { VariableMatch, VariableUsageMatch } from '../../utils/parsers/types'
+
+export default class Usages extends Base {
+    static hidden = false
+
+    static description = 'Print all DevCycle variable usages in the current version of your code.'
+    static examples = [
+        '<%= config.bin %> <%= command.id %>',
+        '<%= config.bin %> <%= command.id %> ' +
+        '--match-pattern javascript="dvcClient\\.variable\\(\\s*["\']([^"\']*)["\']"',
+    ]
+
+    static flags = {
+        ...Base.flags,
+        'include': Flags.string({
+            description: 'Files to include when scanning for usages. By default all files are included. ' +
+                'Accepts multiple glob patterns.',
+            multiple: true
+        }),
+        'exclude': Flags.string({
+            description: 'Files to exclude when scanning for usages. By default all files are included. ' +
+                'Accepts multiple glob patterns.',
+            multiple: true
+        }),
+        'client-name': ClientNameFlag,
```

Comment:
it seems like there's some potential to have a "code insights" base class here that the diff and usage commands can extend from, which itself extends from the base class?